aws/resource_aws_appsync_schema.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"errors"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appsync"
+	"github.com/hashicorp/terraform/helper/schema"
+	"time"
+)
+
+func resourceAwsAppsyncSchema() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppsyncSchemaPut,
+		Read:   resourceAwsAppsyncSchemaNil,
+		Update: resourceAwsAppsyncSchemaPut,
+		Delete: resourceAwsAppsyncSchemaNil,
+
+		Schema: map[string]*schema.Schema{
+			"api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"definition": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppsyncSchemaPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appsyncconn
+
+	apiId := d.Get("api_id").(string)
+	d.SetId(apiId)
+
+	if d.HasChange("definition") {
+		input := &appsync.StartSchemaCreationInput{
+			ApiId:      aws.String(apiId),
+			Definition: ([]byte)(d.Get("definition").(string)),
```

Comment:
Since this attribute is `[]byte`, do we need to worry about actual binary data? Terraform's state currently requires UTF-8 compatibility otherwise it could cause corruption.